parsely/src/test/java/com/parsely/parselyandroid/EngagementManagerTest.kt

```diff
@@ -0,0 +1,158 @@
+package com.parsely.parselyandroid
+
+import androidx.test.core.app.ApplicationProvider
+import java.util.Calendar
+import java.util.TimeZone
+import java.util.Timer
+import org.assertj.core.api.AbstractLongAssert
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.within
+import org.assertj.core.api.Assertions.withinPercentage
+import org.assertj.core.api.MapAssert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+private typealias Event = MutableMap<String, Any>
+
+@RunWith(RobolectricTestRunner::class)
+@Suppress("UNCHECKED_CAST")
```

Comment:
Minor (🔍): Consider moving this suppression closer to the warning, that is, instead of having it here for the class class, to only apply that to the `isCorrectEvent(...)` extension function.